solidity/random-beacon/contracts/ReimbursmentPool.sol

```diff
@@ -0,0 +1,127 @@
+        /* solhint-enable avoid-low-level-calls */
+        require(sent, "Failed to refund Ether");
+    }
+
+    /// @notice Authorize a contract that can interact with this reimbursment pool.
+    ///         Can be authorized by the governance only.
+    /// @param _contract Authorized contract.
+    function authorize(address _contract) external onlyOwner {
+        isAuthorized[_contract] = true;
+    }
+
+    /// @notice Unauthorize a contract that was previously authorized to interact
+    ///         with this reimbursment pool. Can be unauthorized by the
+    ///         governance only.
+    /// @param _contract Authorized contract.
+    function unauthorize(address _contract) external onlyOwner {
+        delete isAuthorized[_contract];
+    }
+
+    /// @notice Setting a static gas cost for executing a transaction. Can be set
+    ///         by the governance only.
+    /// @param _staticGas Static gas cost.
+    function setStaticGas(uint256 _staticGas) external onlyOwner {
+        staticGas = _staticGas;
+
+        emit StaticGasUpdated(_staticGas);
+    }
+
+    /// @notice Setting a max gas price for transactions. Can be set by the
+    ///         governance only.
+    /// @param _maxGasPrice Max gas price used to reimburse tx submitters.
+    function setMaxGasPrice(uint256 _maxGasPrice) external onlyOwner {
+        maxGasPrice = _maxGasPrice;
+
+        emit MaxGasPriceUpdated(_maxGasPrice);
+    }
+
+    /// @notice Withdraws ETH amount from this pool which are sent to a given
+    ///         address. Can be set by the governance only.
+    /// @param amount Amount to withdraw from the pool.
+    /// @param receiver An address where ETH is sent.
+    function withdraw(uint256 amount, address receiver) external onlyOwner {
+        require(
+            address(this).balance >= amount,
+            "Insufficient contract balance"
+        );
+        require(receiver != address(0), "Receiver's address cannot be zero");
+
+        /* solhint-disable avoid-low-level-calls */
+        // slither-disable-next-line low-level-calls
+        (bool sent, ) = receiver.call{value: amount}("");
+        /* solhint-enable avoid-low-level-calls */
+        require(sent, "Failed to send Ether");
+    }
+
+    /// @notice Withdraws all ETH from this pool which are sent to a given
+    ///         address. Can be set by the governance only.
+    /// @param receiver An address where ETH is sent.
+    function withdrawAll(address receiver) external onlyOwner {
+        require(address(this).balance > 0, "Nothing to withdraw");
+        require(receiver != address(0), "Receiver's address cannot be zero");
+
+        /* solhint-disable avoid-low-level-calls */
+        // slither-disable-next-line low-level-calls
+        (bool sent, ) = receiver.call{value: address(this).balance}("");
+        /* solhint-enable avoid-low-level-calls */
+        require(sent, "Failed to send Ether");
+    }
```

Comment:
Do we really need this function? If the answer is yes, maybe we could simply call the `withdraw`?

```suggestion
    function withdrawAll(address receiver) external onlyOwner {
        withdraw(address(this).balance, receiver);
    }
```

Reply:
I think we need it so that we do not have to race to call `withdraw(uint256 amount, address receiver)` with transactions using the pool. See that it fails if the amount does not match.

Reply:
Simplified a bit [59b67ee](https://github.com/keep-network/keep-core/pull/2826/commits/59b67ee770f0031653bdc57cda01165da6c9e48c)